sound/soc/soc-core.c

```diff
@@ -1658,33 +1679,35 @@ int snd_soc_runtime_set_dai_fmt(struct snd_soc_pcm_runtime *rtd,
 	 * Flip the polarity for the "CPU" end of a CODEC<->CODEC link
 	 * the component which has non_legacy_dai_naming is Codec
 	 */
-	if (cpu_dai->component->driver->non_legacy_dai_naming) {
-		unsigned int inv_dai_fmt;
+	for_each_rtd_cpu_dai(rtd, i, cpu_dai) {
+		if (cpu_dai->component->driver->non_legacy_dai_naming) {
```

Comment:
> what is the 'legacy_dai_naming' and can it be removed now?

@plbossart No, I don't think it can be removed now. Codec will be registered to cpu dai in CODEC <-> CODEC link, and we can use the non_legacy_dai_naming flag to know that is codec.